Sources/Overload/OvEditor/src/OvEditor/Panels/MenuBar.cpp

```diff
@@ -197,8 +200,90 @@ void OvEditor::Panels::MenuBar::CreateSettingsMenu()
 
 void OvEditor::Panels::MenuBar::CreateLayoutMenu() 
 {
-	auto& layoutMenu = CreateWidget<MenuList>("Layout");
-	layoutMenu.CreateWidget<MenuItem>("Reset").ClickedEvent += EDITOR_BIND(ResetLayout);
+	auto& layoutMenuList = CreateWidget<MenuList>("Layout");
+
+	auto& saveMenuItem = layoutMenuList.CreateWidget<MenuItem>("Save");
+	saveMenuItem.ClickedEvent += EDITOR_BIND(SaveCurrentLayout);
+
+	auto& saveNewMenuList = layoutMenuList.CreateWidget<MenuList>("Save New");
+	auto& layoutInputText = saveNewMenuList.CreateWidget<InputFields::InputText>("Layout Name");
+	layoutInputText.selectAllOnClick = true;
+	layoutInputText.EnterPressedEvent += [this](std::string p_input)
+	{
+		if (p_input.empty())
+			return;
+
+		auto& UIManager = OvCore::Global::ServiceLocator::Get<OvUI::Core::UIManager>();
+		std::string layoutsPath = UIManager.GetLayoutsPath();
+		EDITOR_EXEC(DelayAction(std::bind(&OvUI::Core::UIManager::SaveLayout, &UIManager, layoutsPath + p_input + ".ini"), 1));
+	};
+
+	auto& loadMenuList = layoutMenuList.CreateWidget<MenuList>("Load");
+
+	loadMenuList.ClickedEvent += [&]
+	{
+		loadMenuList.RemoveAllWidgets();
+
+		auto& UIManager = OvCore::Global::ServiceLocator::Get<OvUI::Core::UIManager>();
+		std::string layoutsPath = UIManager.GetLayoutsPath();
+
+		for (const auto& entry : std::filesystem::directory_iterator(layoutsPath))
+		{
+			if (entry.is_regular_file() && entry.path().extension() == ".ini")
+			{
+				std::shared_ptr<std::string> layoutFileName = std::make_shared<std::string>(entry.path().filename().string());
```

Comment:
Any reason why we are allocating this string dynamically, and with a `shared_ptr`? I don't see any other code path that shares the ownership of this memory location